pkg/queue/stats_reporter_test.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queue
+
+import (
+	"errors"
+	"testing"
+
+	"go.opencensus.io/stats/view"
+)
+
+const (
+	namespace = "default"
+	config    = "helloworld-go"
+	revision  = "helloworld-go-00001"
+)
+
+func TestNewStatsReporter_negative(t *testing.T) {
+	tests := []struct {
+		name      string
+		errorMsg  string
+		result    error
+		namespace string
+		config    string
+		revision  string
+	}{
+		{
+			"Empty_Namespace_Value",
+			"Expected namespace empty error",
+			errors.New("Namespace must not be empty"),
+			"",
+			config,
+			revision,
+		},
+		{
+			"Empty_Config_Value",
+			"Expected config empty error",
+			errors.New("Config must not be empty"),
+			namespace,
+			"",
+			revision,
+		},
+		{
+			"Empty_Revision_Value",
+			"Expected revision empty error",
+			errors.New("Revision must not be empty"),
+			namespace,
+			config,
+			"",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := NewStatsReporter(test.namespace, test.config, test.revision); err.Error() != test.result.Error() {
```

Comment:
Can you not compare the errors directly?

Reply:
What do you propose?

Reply:
I was wodndering if you need to extract the error message to compare or whether you could do `err != test.result` directly.

Reply:
I believe that does a comparison of object references not necessarily the error itself. Long story short I had tried that before myself and discovered it's not the way to compare error objects. Unfortunately I don't see a comapreto or equals method in the interface either so kinda stuck with this method of string comparison.

Reply:
Ah, you're right. I think I was fooled by the way error handling is done usually (via constant references). Of course it works there.

Fine with me!